src/framework/components/app/core.clj

```diff
@@ -65,22 +66,25 @@
        (map context)
        (map #(fn [x] (% x)))))
 
-(defrecord App
-  [config router db]
+(defn init-acl
+  [this config]
+  (acl-builder/init this config))
 
+(defrecord App
+  [config acl-cfg router db]
   component/Lifecycle
   (stop [this] this)
   (start [this]
          (assoc this
            :handler
-
            (fn [http-request]
              (->
                (apply m/>>=
                  (concat
                    [(xiana.core/ok (create-empty-state))
                     (fn [x] (add-deps x {:router router, :db db}))
-                    (fn [x] (add-http-request x http-request))]
+                    (fn [x] (add-http-request x http-request))
+                    (fn [this] (init-acl this acl-cfg))]
```

Comment:
you can access `acl-cfg` using `(-> this :acl-cfg)`, so no need for acl-cfg to be a component